src/llama_stack_client/_client.py

```diff
@@ -123,7 +125,14 @@ def __init__(
         _strict_response_validation: bool = False,
         provider_data: Mapping[str, Any] | None = None,
     ) -> None:
-        """Construct a new synchronous llama-stack-client client instance."""
+        """Construct a new synchronous llama-stack-client client instance.
+
+        This automatically infers the `api_key` argument from the `LLAMA_STACK_API_KEY` environment variable if it is not provided.
+        """
+        if api_key is None:
+            api_key = os.environ.get("LLAMA_STACK_API_KEY")
```

Comment:
We should name this LLAMA_STACK_CLIENT_API_KEY 

so it has the same prefix as LLAMA_STACK_CLIENT_BASE_URL.